### ISSUE TITLE
Additional updates to panel components and page link components

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#798]
Filing this issue to follow up with suggested changes in https://github.com/primer/view_components/issues/680.

Broadly summarizing....
#### Suggestions related to our tabbed panel components (`TabPanels` and `UnderlinePanels`):

- programmatically determine `tabindex` based on panel content
- update accessibility guidance and examples so panels have heading. potentially update API to require a heading for each panel.

#### Suggestions related to our page link components (`TabNav` and `UnderlineNav`):

- introduce heading slot which should be required for the `nav` case. we may also want to introduce an argument that allows this to be visually hidden.
- update how we label`nav` and `ul`

In addition, we should follow up with the discussion with design around refining use cases for these components. We may end up deprecating one of `TabNav`, `UnderlineNav`, and one of `TabPanels`, `UnderlinePanels`.

#### Relevant discussions
- https://github.com/github/primer/issues/265
- https://github.com/primer/view_components/pull/785
